certbot/certbot/_internal/client.py

```diff
@@ -405,18 +441,20 @@ def obtain_certificate(self, domains, old_keypath=None):
             cert, chain = self.obtain_certificate_from_csr(csr, orderr)
             return cert, chain, key, csr
 
-    def _get_order_and_authorizations(self, csr_pem: str,
+    def _get_order_and_authorizations(self, csr_pem: bytes,
                                       best_effort: bool) -> messages.OrderResource:
         """Request a new order and complete its authorizations.
 
-        :param str csr_pem: A CSR in PEM format.
+        :param bytes csr_pem: A CSR in PEM format.
         :param bool best_effort: True if failing to complete all
             authorizations should not raise an exception
 
         :returns: order resource containing its completed authorizations
         :rtype: acme.messages.OrderResource
 
         """
+        if not self.acme:
+            raise errors.Error("ACME client is not set.")
```

Comment:
Indeed I consider `assert` more as a debug instruction which can be disabled in theory using `python -O`. This is my personal feeling indeed, but I always find it weird to have them in functional code. Now it is true that effectively `self.acme` is never `None` at the end, and so we really just want to help mypy here. However given that we claim that `self.acme` is optional, it may happen in the future that some developments put a `None` with good or bad reason. I would feel more comfortable with an assertion that cannot be disabled implicitely by the python runtime configuration.

Reply:
OK, sounds good to me.